File: app/src/main/java/pl/mosenko/sunnypodlaskie/util/RawResourceUtil.java
package pl.mosenko.sunnypodlaskie.util;

import android.content.Context;
import android.support.annotation.NonNull;
import android.util.Log;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStream;
import java.io.InputStreamReader;

import static android.icu.lang.UCharacter.GraphemeClusterBreak.T;

/**
 * Created by syk on 27.05.17.
 */

public class RawResourceUtil {
    private static final String TAG = RawResourceUtil.class.getSimpleName();
    private RawResourceUtil() {}

    public static String readRawTextFile(@NonNull Context context, @NonNull int resId) {
        InputStream inputStream = context.getResources().openRawResource(resId);
        try {
            BufferedReader reader = new BufferedReader(new InputStreamReader(inputStream));
            StringBuilder sb = new StringBuilder();
            String line;
            while ((line = reader.readLine()) != null) {
                sb.append(line).append("\n");
            }
            reader.close();
            return sb.toString();
        } catch (IOException e) {
            Log.e(TAG, "Exceptino during reading text from raw file.", e);
        }
        return null;
    }
}

File: app/src/main/java/pl/mosenko/sunnypodlaskie/util/RawResourceUtil.kt
package pl.mosenko.sunnypodlaskie.util

import android.content.Context
import android.util.Log
import pl.mosenko.sunnypodlaskie.BuildConfig
import java.io.BufferedReader
import java.io.IOException
import java.io.InputStream
import java.io.InputStreamReader

/**
 * Created by syk on 27.05.17.
 */
object RawResourceUtil {
    private val TAG = RawResourceUtil::class.java.simpleName
    fun readRawTextFile(context: Context, resId: Int): String? {
        val inputStream = context.resources.openRawResource(resId)
        return readTextFromInputStream(inputStream)
    }

    fun readTextFromInputStream(inputStream: InputStream?): String? {
        try {
            val reader = BufferedReader(InputStreamReader(inputStream))
            val sb = StringBuilder()
            var line: String?
            while (reader.readLine().also { line = it } != null) {
                sb.append(line).append("\n")
            }
            reader.close()
            return sb.toString()
        } catch (e: IOException) {
            if (BuildConfig.DEBUG) {
                Log.e(TAG, "Exception during reading text from the raw file.", e)
            }
        }
        return null
    }
}
